### ISSUE TITLE
Autocomplete for import behaves badly

### ISSUE THREAD

**Issue by brainsik** [DamnWidget/anaconda#89]
**NOTE**: Workaround to solve the issue https://github.com/DamnWidget/anaconda/issues/89#issuecomment-68853747

When doing an import statement, after the space the autocomplete pops up. However, typing anything causes the popup to disappear. If you press tab after typing some of the module you want to import, the autocomplete popup returns to complete the module name, but it's inserted after what you've already typed plus a tab. E.g., it'll look like this:

``` python
import datet<tab>datetime
```

Hoping that's clear. This is the only thing I've seen behave like this. Using the `from` form works fine.

**Comment by hballard** [comment]
I have the exact same problem as well.  I'll subscribe to this thread, in case you find a solution.  Thanks.

**Comment by DamnWidget** [comment]
Hi guys.

Sublime Text 3 Python default package decide to cancel the autocompletion when some words are detected (for example def or class) but it also decides to cancel it with the word "import".

To fix that behavior and make ST3 use the anaconda's completion create a new `Python` directory in your `Packages` directory and copy the contents of the file `https://raw.githubusercontent.com/DamnWidget/anaconda/master/Completion%20Rules.tmPreferences` there.

It will solve the issue.

P.S: I will add this to the README in the next release.

**Comment by hballard** [comment]
Thanks.  That worked for me.

**Comment by zxcbryanfury** [comment]
@DamnWidget  In my situation now, when doing an import statement, after the space the autocomplete pops up but "module" didn't appear in the pop-up list, even after I used your solution.
Then I reinstalled the sublime and used your solution again, this time it works, thank you very much ,anaconda is the best sublime python plugin I ever seen.

**Comment by DamnWidget** [comment]
@zxcbryanfury glad to see that your problem is gone. Thank you, at least is the best sublime python plugin that I ever wrote :)

**Comment by hballard** [comment]
Agree with zxcbryanfury...best plugin on Sublime Text!!  No need for a heavy IDE now!

**Comment by zxcbryanfury** [comment]
Ah ha, I figureed out something. In my last comment, I said " 'module' didn't appear in the pop-up list, even after I used your solution. Then I reinstalled the sublime and used your solution again, this time it works".  The truth is that, the python file which I'm editing is in the same directory with a python file which has a non-English file name (such as Chinese) , so when I use import, "module" won't appear in the pop-up list.

**Comment by brainsik** [comment]
@DamnWidget As OP, confirming your workaround solved my problem. Thanks for enduring this one. :-)

**Comment by DamnWidget** [comment]
It took a while to figure it out why the Completion Rules weren't working :)

**Comment by pradyun** [comment]
Shouldn't this be closed now that it's solved? :confused:

**Comment by DamnWidget** [comment]
I don't think so. I maintained this open as this is a Sublime Text 3 problem itself, so it can't be solved directly from anaconda's implementation. A workaround is not a fix.

**Comment by blink1073** [comment]
I made the file `~/.config/sublime-text-3/Packages/Python/Completion Rules.tmPreferences` with the given contents.  I tried before and after reinstalling Sublime Text 3, to no avail.  

I am using Ubuntu 14.04 with Sublime 3065.  I still see the `import date\tdatetime` behaviour.

**Comment by DamnWidget** [comment]
Thats curious as the thread OP confirmed that the workaround fixed his problem, not sure how to help you here.

Btw, if you are not using autocompletion in [dot] yet I suggest you to do it: http://damnwidget.github.io/anaconda/IDE/#toc_3

**Comment by blink1073** [comment]
Hmm, didn't work on Windows 7 with 3065 either.  Autocomplete on dot is a welcome addition.  Thanks for that and for this wonderful plugin @DamnWidget!

**Comment by Grokzen** [comment]
@DamnWidget Workaround do not work with ST3 build 3086

**Comment by DamnWidget** [comment]
Are you sure? I just tested it in a fresh ST3 3086 installation in Ubuntu 15.04 and it works fine.

**Comment by Grokzen** [comment]
@DamnWidget I added it to the wrong python folder, i am using PythonImproved and after i added it there along with the normal Python folder, it started to work as expected and auto completions for imports work. I changed my comment from `do not work` to `works` after i noticed that :p

**Comment by DamnWidget** [comment]
Ok :)

On 25 May 2015 at 09:40, Grok <notifications@github.com> wrote:

> @DamnWidget <https://github.com/DamnWidget> I added it to the wrong
> python folder, i am using PythonImproved and after i added it there along
> with the normal Python folder, it started to work as expected and auto
> completions for imports work. I changed my comment from do not work to
> works after i noticed that :p
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DamnWidget/anaconda/issues/89#issuecomment-105169802>.
>



-- 
Join the Free Software Foundation and become Free as in Freedom

**Comment by osirisgothra** [comment]
As far as the workaround goes and people saying they need to re-install ST3, etc -- could that be because of something lingering  /.config/sublime-text-3/Cache caused this side effect, rather than some mystery reason (found by looking into the magic :8ball:)? And, if there is a bug in ST3 causing this, has anyone opened it on their bug management site? If so, it would be good to provide a link here, since bugs effectively get fixed faster when more people get together and sign on and report that (or click some check box that says)  'this bug affects me as well' or something to that effect that will eventually effect change on their part.

**Comment by osirisgothra** [comment]
oh yeah, forgot to mention the whole point: i wrote that because I found that workaround file was already in _my_ user cache, here's a cut-n-paste of the line:

``` ... ```
``` ~/.config/sublime-text-3/Cache/Python/Completion Rules.tmPreferences.cache ```
``` ~/.config/sublime-text-3/Cache/Python/Miscellaneous.tmPreferences.cache ```
``` ~/.config/sublime-text-3/Cache/Python/Symbol List Hide Decorator.tmPreferences.cache ```
``` ~/.config/sublime-text-3/Cache/Python/Symbol List.tmPreferences.cache ```
``` ... ```

Mind you, even _with_ the Python/Completion Rules.tmPreferences that you provided us with in it's proper place _and_ after a _full_ restart of ST3, the problem persisted without change.

After that, **I exited ST3** and then I purged my entire ```~/.config/sublime-text-3/Cache``` directory even the directory itself, then recreated it (probably wasn't needed). I then restarted and guess what, everything worked perfectly from that point on. Turned out that stupid Cache directory was also the source of some _other_ woes I faced with ST3 -- thus nobody should have to reinstall ST3 entirely anymore or even remove their .config/sublime-text-3 directory! :) Hope that helped!!

**Comment by DamnWidget** [comment]
README.md updated https://github.com/DamnWidget/anaconda/blob/master/README.md#auto-complete-for-import-behaves-badly
